### PR TITLE
Rdstdin pass

### PR DESCRIPTION
**Pull request by def-**
Example usage:

```nim
import rdstdin

let pass = readPasswordFromStdin "What is the password? "
echo pass
```

Only tested on Linux. Code adapted from this one by Apollo: http://forum.nim-lang.org/t/758

### CODE REVIEW

File: lib/posix/termios.nim [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+const 
+  TCSANOW* = 0
+  TCSADRAIN* = 1
+  TCSAFLUSH* = 2
+
+# Compare a character C to a value VAL from the `cc' array in a
+#   `struct termios'.  If VAL is _POSIX_VDISABLE, no character can match it.  
+
+template CCEQ*(val, c: expr): expr = 
+  ((c) == (val) and (val) != POSIX_VDISABLE)
```
Comment (Araq) [inline lib/posix/termios.nim]: Get rid of the (). Nim is not C.

File: lib/posix/termios.nim [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+type 
+  Speed* = cuint
+  Tcflag* = cuint
+
+const 
+  NCCS* = 32
+
+type 
+  Termios* = object {.importc: "struct termios", header: "termios.h>", final, pure.}
```
Comment (Araq) [inline lib/posix/termios.nim]: Typo: '<' missing

File: lib/posix/termios.nim [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+# Compare a character C to a value VAL from the `cc' array in a
+#   `struct termios'.  If VAL is _POSIX_VDISABLE, no character can match it.  
+
+template CCEQ*(val, c: expr): expr = 
+  ((c) == (val) and (val) != POSIX_VDISABLE)
+
+# Return the output baud rate stored in *TERMIOS_P.  
+
+proc cfgetospeed*(termios: ptr Termios): Speed {.importc: "cfgetospeed",
```
Comment (Araq) [inline lib/posix/termios.nim]: cfGetToSpeed is a nicer name and with nimfmt relevant as nimfmt will make the spelling consistent with the defintion. This applies to the other names too.

File: lib/impure/rdstdin.nim [hunk 4]
```diff
@@ -31,9 +31,27 @@ when defined(Windows):
...
+          break
+       of '\b':
+          password.setLen(result.len - 1)
+       else:
+          password.add(c)
+
 else:
-  import readline, history
-    
+  import readline, history, termios, unsigned
```
Comment (Araq) [inline lib/impure/rdstdin.nim]: I guess it's fine in rdstdin, but perhaps a seperate password module is better? Dunno.

Comment (def-) [inline lib/impure/rdstdin.nim]: I think the rdstdin module is pretty small as it is. Didn't want to make yet another small module with just 2 procs.

Comment (dom96) [inline lib/impure/rdstdin.nim]: Yes, keep it in the rdstdin module.

File: lib/posix/termios.nim [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+
+const 
+  TCSANOW* = 0
+  TCSADRAIN* = 1
+  TCSAFLUSH* = 2
+
+# Compare a character C to a value VAL from the `cc' array in a
+#   `struct termios'.  If VAL is _POSIX_VDISABLE, no character can match it.  
+
+template CCEQ*(val, c: expr): expr =
```
Comment (Araq) [inline lib/posix/termios.nim]: Oh and this should be 'cceq'. Sorry, I didn't complain earlier.

**Pull request merged**